src/check.c

```diff
@@ -166,7 +179,10 @@ static void tcase_free(TCase * tc)
     check_list_free(tc->ch_sflst);
     check_list_free(tc->unch_tflst);
     check_list_free(tc->ch_tflst);
-
+    if (tc->tags) {
```

Comment:
nit pick, but please use the same style which is used elsewhere, namely place the { on the next line, and always use {} even if there is a one line body, for example:
```
if(condition)
{
   do_something();
}
```